Adapt GetNearestTrack() method

martenole opened the pull request:
- also add copy constructor and assignment operator
- add streamer as member to the class
- fix a few typos

martenole committed:
ATO-438: Adapt GetNearestTrack function

martenole committed:
ATO-438: Add streaming of event variables

martenole committed:
ATO-438: Additional statistics for debug tree

miranov25 commented:
Hallo Ole.

Thank you for request. Lets review the code. Maybe best in person.
* You added constructors - that is fine. 
* For the GetNearestTrack you modified the interface.
   * Could you update than the description ?
   * if I understand correctly you are using unly static function of tracker? Is it than needed?
   * Could we parameterize status conditions for nearest tracks ?

martenole committed:
ATO-438: Add description for GetNearestTrack()

- fix global event id calculation
- track type as additional argument

martenole commented:
Hi Marian,

I have added now the description to the GetNearestTrack method and added the types of the probe track and the track to look for as parameters.

The status condition parametrization is still missing.

For the tracker I do not know how to make it static, since it needs to be initialized correctly for the given run with B-field, misalignment etc.

Cheers,
Ole

miranov25 commented:
Closing old pull request
